Lib/ufoLib/glifLib.py

```diff
@@ -800,42 +803,30 @@ def validateLayerInfoVersion3Data(infoData):
 # GLIF Tree Support
 # -----------------
 
-def _stripGlyphXMLTree(nodes):
-	for node in nodes:
-		if len(node) != 3:
-			raise GlifLibError("Invalid GLIF structure.")
-		element, attrs, children = node
-		# "lib" is formatted as a plist, so we need unstripped
-		# character data so we can support strings with leading or
-		# trailing whitespace. Do strip everything else.
-		recursive = (element != "lib")
-		stripCharacterData(children, recursive=recursive)
-
 def _glifTreeFromFile(aFile):
-	tree = buildTree(aFile, stripData=False)
-	stripCharacterData(tree[2], recursive=False)
-	if tree[0] != "glyph":
+	root = ElementTree.parse(aFile).getroot()
+	if root.tag != "glyph":
+		raise GlifLibError("The GLIF is not properly formatted.")
+	return root
+
+def _glifTreeFromString(aString):
+	root = ElementTree.fromstring(aString)
+	if root.tag != "glyph":
 		raise GlifLibError("The GLIF is not properly formatted.")
-	_stripGlyphXMLTree(tree[2])
-	return tree
+	return root
 
 def _readGlyphFromTree(tree, glyphObject=None, pointPen=None, formatVersions=(1, 2)):
-	# quick format validation
-	formatError = False
-	if len(tree) != 3:
-		formatError = True
-	else:
-		if tree[0] != "glyph":
-			formatError = True
-	if formatError:
-		raise GlifLibError("GLIF data is not properly formatted.")
 	# check the format version
-	formatVersion = tree[1].get("format", "undefined")
+	formatVersion = tree.get("format")
+	if formatVersion is None:
+		raise GlifLibError("Unspecified format version in GLIF.")
 	try:
 		v = int(formatVersion)
 		formatVersion = v
 	except ValueError:
 		pass
+	if formatVersion not in formatVersions:
+		raise GlifLibError("Forbidden GLIF format version: %s" % formatVersion)
```

Comment:
Well it's not necessarily unknown given that formatVersions is an argument that serves to forbid certain versions (mostly to forbid GLIF2 in UFO2). Sure you could have formatVersion=3 which is unknown but here is mostly about limiting GLIF versions for older versions of the standard, I reckon.